### PR TITLE
9634 Dev. Guide checklist for the tasks required when making a field multivalued

### LINKED ISSUE #9634: Document the checklist of all the tasks necessary when making an existing single field multiple
The recently merged PRs that made the formerly single value-only fields "productionPlace" and "series" multiple (#9254 and #9256 respectively), introduced A LOT of problems in the export and import code that the core team had to spend a lot of time address after the fact. Some were not even caught in time (for example, the "all fields" sample json file that's linked in the v5.13 (latest release as of now) guide ([dataset-create-new-all-default-fields.json](https://guides.dataverse.org/en/5.13/_downloads/4e04c8120d51efab20e480c6427f139c/dataset-create-new-all-default-fields.json)) has an entry for `productionPlace` that is encoded as single-only, thus making the json un-importable once the citation block is updated as part of a v5.13 install or upgrade. 

It would not be fair to blame the developer for the oversights. It's not obvious or intuitive that all these changes need to be made and where. And we apparently don't have enough automated tests to alert a  developer of all these problems. So we need to document the checklist of all the required tasks in the developers guide. And then if (God forbid) anyone ever needs to make another existing single-only field multiple, it would be easy to make it all 100% responsibility of the developer. 

(It says something about our metadata import/export framework, that it requires all these explicit code fixes to accommodate such a change - ideally, all that behavior could be made to be derived from the metadata block configuration (?)... but that is absolutely outside of the scope of this PR). 

We could add a couple of extra integration tests though to detect some of the problems in question. For example, it would be trivial to add an import test for our provided sample file mentioned above to ensure that it stays importable. That can be handled as part of this issue as well.

### PR DESCRIPTION
**Pull request by landreev**
**What this PR does / why we need it**:

A documentation-only PR. Some of the tasks involved are not immediately obvious to an outside contributor, and this resulted in minor problems in the past couple of years when some core fields had to be made multiple. 

**Which issue(s) this PR closes**:

Closes #9634

**Special notes for your reviewer**:

Most of the information we provide on customizing metadata blocks is under the Admin Guide. This specific list is oriented towards a contributor making a PR, so I created it under the Developer Guide, and linked to it from the Metadata Blocks section of the Admin. 

**Suggestions on how to test this**:

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**:

**Additional documentation**:

### CODE REVIEW

File: doc/sphinx-guides/source/developers/metadatablocksdev.rst
Comment (pdurbin) [inline doc/sphinx-guides/source/developers/metadatablocksdev.rst:1]: A high-level observation: we already have a good amount of "dev" stuff under the main metadata customization page. We talk about using Docker, dropping your database and starting over if necessary, etc. Should we move that content into this dev page?

**Review by pdurbin — COMMENTED** [review]
Just a quick comment about dev-related content.

File: doc/sphinx-guides/source/developers/metadatablocksdev.rst
Comment (landreev) [reply doc/sphinx-guides/source/developers/metadatablocksdev.rst:1]: No strong opinion on that part - just say the word and I'll move it. 
I can see the advantage in not splitting information between different parts of the guide.

Comment (sekmiller) [reply doc/sphinx-guides/source/developers/metadatablocksdev.rst:1]: Makes sense to me to keep it all in one place.

Comment (landreev) [reply doc/sphinx-guides/source/developers/metadatablocksdev.rst:1]: OK, done. That was a good call, it simplified it and reduced the amount of duplicated language.

**Review by sekmiller — APPROVED** [review]

File: doc/sphinx-guides/source/admin/metadatacustomization.rst [hunk 1]
```diff
@@ -648,6 +648,28 @@ Alternatively, you are welcome to request "edit" access to this "Tips for Datave
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As a recent real life example, a few fields from the Citation and Social Science block were changed to support multiple values, in order to accommodate specific needs of some community member institutions. A PR for one of these fields, ``alternativeTitle`` from the Citation block is linked below. Each time a number of code changes, plus some changes in the sample metadata files in the Dataverse code tree had to be made. The checklist below is to help another developer in the event that a similar change becomes necessary in the future. Note that some of the steps below may not apply 1:1 to a different metadata field, depending on how it is exported and imported in various formats by Dataverse. It may help to consult the PR `#9440 <https://github.com/IQSS/dataverse/pull/9440/files>`_ as a specific example of the changes that had to be made for the ``alternativeTitle`` field. 
+
+- Change the value from ``FALSE`` to ``TRUE`` in the ``alowmultiples`` column of the .tsv file for the block.
+- Change the value of the ``multiValued`` attribute for the search field in the Solr schema (``conf/solr/9.3.0/schema.xml`` as of writing this).
```
Comment (pdurbin) [inline doc/sphinx-guides/source/admin/metadatacustomization.rst:662]:
```suggestion
- Change the value of the ``multiValued`` attribute for the search field in the Solr schema (``conf/solr/x.x.x/schema.xml``).
```

File: doc/sphinx-guides/source/admin/metadatacustomization.rst [hunk 2]
```diff
@@ -648,6 +648,28 @@ Alternatively, you are welcome to request "edit" access to this "Tips for Datave
...
+
+Making a Field Multi-Valued
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As a recent real life example, a few fields from the Citation and Social Science block were changed to support multiple values, in order to accommodate specific needs of some community member institutions. A PR for one of these fields, ``alternativeTitle`` from the Citation block is linked below. Each time a number of code changes, plus some changes in the sample metadata files in the Dataverse code tree had to be made. The checklist below is to help another developer in the event that a similar change becomes necessary in the future. Note that some of the steps below may not apply 1:1 to a different metadata field, depending on how it is exported and imported in various formats by Dataverse. It may help to consult the PR `#9440 <https://github.com/IQSS/dataverse/pull/9440/files>`_ as a specific example of the changes that had to be made for the ``alternativeTitle`` field. 
+
+- Change the value from ``FALSE`` to ``TRUE`` in the ``alowmultiples`` column of the .tsv file for the block.
```
Comment (pdurbin) [inline doc/sphinx-guides/source/admin/metadatacustomization.rst:661]:
```suggestion
- Change the value from ``FALSE`` to ``TRUE`` in the ``allowmultiples`` column of the .tsv file for the block.
```

File: doc/sphinx-guides/source/admin/metadatacustomization.rst [hunk 3]
```diff
@@ -648,6 +648,28 @@ Alternatively, you are welcome to request "edit" access to this "Tips for Datave
...
+- Change the value of the ``multiValued`` attribute for the search field in the Solr schema (``conf/solr/9.3.0/schema.xml`` as of writing this).
+- Modify the DDI import code (``ImportDDIServiceBean.java``) to support multiple values. (you may be able to use the change in the PR above as a model.)
```
Comment (pdurbin) [inline doc/sphinx-guides/source/admin/metadatacustomization.rst:663]:
```suggestion
- Modify the DDI import code (``ImportDDIServiceBean.java``) to support multiple values. (You may be able to use the change in the PR above as a model.)
```

File: doc/sphinx-guides/source/admin/metadatacustomization.rst [hunk 4]
```diff
@@ -648,6 +648,28 @@ Alternatively, you are welcome to request "edit" access to this "Tips for Datave
...
+- Modify the DDI export utility (``DdiExportUtil.java``).
+- Modify the OpenAire export utility (``OpenAireExportUtil.java``).
+- Modify the following JSON source files in the Dataverse code tree to actually include multiple values for the field (two should be quite enough!): ``scripts/api/data/dataset-create-new-all-default-fields.json``, ``src/test/java/edu/harvard/iq/dataverse/export/dataset-all-defaults.txt``, ``src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-finch1.json`` and ``src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-create-new-all-ddi-fields.json``. (These are used as examples for populating datasets via the import API and by the automated import and export code tests).
+- Similarly modify the following XML files that are used by the DDI export code tests: ``src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-finch1.xml`` and ``src/test/java/edu/harvard/iq/dataverse/export/ddi/exportfull.xml``.
+- Make sure all the automated Unit and Integration tests are passing.
+- Write a short release note to announce the change in the upcoming release.
+- Make a Pull Request.
```
Comment (pdurbin) [inline doc/sphinx-guides/source/admin/metadatacustomization.rst:670]:
```suggestion
- Make a pull request. 
```

File: doc/sphinx-guides/source/admin/metadatacustomization.rst [hunk 5]
```diff
@@ -648,6 +648,28 @@ Alternatively, you are welcome to request "edit" access to this "Tips for Datave
...
+- Modify the DDI export utility (``DdiExportUtil.java``).
+- Modify the OpenAire export utility (``OpenAireExportUtil.java``).
+- Modify the following JSON source files in the Dataverse code tree to actually include multiple values for the field (two should be quite enough!): ``scripts/api/data/dataset-create-new-all-default-fields.json``, ``src/test/java/edu/harvard/iq/dataverse/export/dataset-all-defaults.txt``, ``src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-finch1.json`` and ``src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-create-new-all-ddi-fields.json``. (These are used as examples for populating datasets via the import API and by the automated import and export code tests).
+- Similarly modify the following XML files that are used by the DDI export code tests: ``src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-finch1.xml`` and ``src/test/java/edu/harvard/iq/dataverse/export/ddi/exportfull.xml``.
+- Make sure all the automated Unit and Integration tests are passing.
+- Write a short release note to announce the change in the upcoming release.
```
Comment (pdurbin) [inline doc/sphinx-guides/source/admin/metadatacustomization.rst:669]: We could add the :ref: thing to https://dataverse-guide--10284.org.readthedocs.build/en/10284/developers/version-control.html#writing-a-release-note-snippet

File: doc/sphinx-guides/source/admin/metadatacustomization.rst [hunk 6]
```diff
@@ -648,6 +648,28 @@ Alternatively, you are welcome to request "edit" access to this "Tips for Datave
...
+- Modify the DDI import code (``ImportDDIServiceBean.java``) to support multiple values. (you may be able to use the change in the PR above as a model.)
+- Modify the DDI export utility (``DdiExportUtil.java``).
+- Modify the OpenAire export utility (``OpenAireExportUtil.java``).
+- Modify the following JSON source files in the Dataverse code tree to actually include multiple values for the field (two should be quite enough!): ``scripts/api/data/dataset-create-new-all-default-fields.json``, ``src/test/java/edu/harvard/iq/dataverse/export/dataset-all-defaults.txt``, ``src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-finch1.json`` and ``src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-create-new-all-ddi-fields.json``. (These are used as examples for populating datasets via the import API and by the automated import and export code tests).
+- Similarly modify the following XML files that are used by the DDI export code tests: ``src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-finch1.xml`` and ``src/test/java/edu/harvard/iq/dataverse/export/ddi/exportfull.xml``.
+- Make sure all the automated Unit and Integration tests are passing.
```
Comment (pdurbin) [inline doc/sphinx-guides/source/admin/metadatacustomization.rst:668]:
```suggestion
- Make sure all the automated unit and integration tests are passing.
```

**Review by pdurbin — COMMENTED** [review]
Some small suggestions.

**Comment by pdurbin** [comment]
I checked with Leonid and he's fine with my proposed changes. I made them and will merge this.

**Pull request merged**